src/http/server_context.rs

```diff
@@ -121,4 +82,84 @@ impl ServerContext {
       env_vars: Arc::new(env),
     }
   }
+
+  #[allow(clippy::too_many_arguments)]
+  fn check_resolver(
+    field: &mut blueprint::FieldDefinition,
+    expr: &Expression,
+    bt: Batch,
+    universal_http_client: Arc<dyn HttpClient>,
+    http2_only_client: Arc<dyn HttpClient>,
+    http_data_loaders: &mut Vec<DataLoader<DataLoaderRequest, HttpDataLoader>>,
+    gql_data_loaders: &mut Vec<DataLoader<DataLoaderRequest, GraphqlDataLoader>>,
+    grpc_data_loaders: &mut Vec<DataLoader<grpc::DataLoaderRequest, GrpcDataLoader>>,
+  ) -> Expression {
+    if let Expression::Unsafe(expr_unsafe) = &expr {
+      match expr_unsafe {
+        Unsafe::Http { req_template, group_by, .. } => {
+          let data_loader = HttpDataLoader::new(
+            universal_http_client.clone(),
+            group_by.clone(),
+            matches!(&field.of_type, ListType { .. }),
+          )
+          .to_data_loader(bt);
+
+          field.resolver = Some(Expression::Unsafe(Unsafe::Http {
+            req_template: req_template.clone(),
+            group_by: group_by.clone(),
+            dl_id: Some(DataLoaderId(http_data_loaders.len())),
+          }));
+
+          http_data_loaders.push(data_loader);
+        }
+
+        Unsafe::GraphQLEndpoint { req_template, field_name, batch, .. } => {
+          let graphql_data_loader = GraphqlDataLoader::new(universal_http_client.clone(), *batch).to_data_loader(bt);
+
+          field.resolver = Some(Expression::Unsafe(Unsafe::GraphQLEndpoint {
+            req_template: req_template.clone(),
+            field_name: field_name.clone(),
+            batch: *batch,
+            dl_id: Some(DataLoaderId(gql_data_loaders.len())),
+          }));
+
+          gql_data_loaders.push(graphql_data_loader);
+        }
+
+        Unsafe::Grpc { req_template, group_by, .. } => {
+          let data_loader = GrpcDataLoader {
+            client: http2_only_client.clone(),
+            operation: req_template.operation.clone(),
+            group_by: group_by.clone(),
+          };
+          let data_loader = data_loader.to_data_loader(bt);
+
+          field.resolver = Some(Expression::Unsafe(Unsafe::Grpc {
+            req_template: req_template.clone(),
+            group_by: group_by.clone(),
+            dl_id: Some(DataLoaderId(grpc_data_loaders.len())),
+          }));
+
+          grpc_data_loaders.push(data_loader);
+        }
+        _ => {}
+      }
+      field.resolver.clone().unwrap()
+    } else if let Expression::Cache(cache) = &expr {
+      let new_expr = Self::check_resolver(
+        field,
+        cache.source(),
+        bt,
+        universal_http_client,
+        http2_only_client,
+        http_data_loaders,
+        gql_data_loaders,
+        grpc_data_loaders,
+      );
+      field.resolver = Some(Expression::Cache(Cache::new(cache.max_age(), Box::new(new_expr))));
+      field.resolver.clone().unwrap()
```

Comment:
I think the call for `check_resolver` will already change the resolver `field` so we don't need to do this again. Could you verify this?